File: deepagents/maaya/test_ws.js
const WebSocket = require('ws');

const ws = new WebSocket('ws://localhost:8000/ws/chat');

ws.on('open', () => {
    console.log('Connected');
    ws.send(JSON.stringify({ type: 'message', content: 'SUPER SECRET KEYWORD' }));
});

ws.on('message', (data) => {
    console.log('Received:', data.toString());
});

ws.on('close', (code, reason) => {
    console.log('Disconnected', code, reason.toString());
});

ws.on('error', console.error);

setTimeout(() => {
    console.log('Timeout reached. Closing.');
    ws.close();
}, 5000);

File: deepagents/maaya/test_ws.py
import asyncio
import websockets
import json

async def test():
    async with websockets.connect("ws://localhost:8000/ws/chat") as ws:
        await ws.send(json.dumps({"type": "message", "content": "SUPER SECRET KEYWORD"}))
        try:
            while True:
                resp = await asyncio.wait_for(ws.recv(), timeout=5.0)
                print("RECV:", resp)
        except asyncio.TimeoutError:
            print("timeout")

asyncio.run(test())
